### DISCUSSION TITLE
Feature request: MIDI link

### DISCUSSION THREAD

**Discussion by reneknuvers** [jamulussoftware/jamulus#2071] [Ideas]
Objective: to stream midi data from one client to the other. This will allow for control of midi devices and is useful for keeping harmonizers in sync. Other uses are a local controllable click track / metronome, control of remote patch changes, display of chords played on a screen, and control of songbook software

Technical realization: MIDI stream has a max baud rate of 31.25kbaud. A jamulus client software should attach to one midi port as available on Jack or whatever the local OS provides as a midi hook. The stream should be transported along the audio to the connected server and redistributed to each client. Per client the midi stream should be 'muted' or aggregated to a new (virtual) midi port for further use by the local OS.

This breaks: nothing as far as I can tell. Implementation should avoid conflicts with current local midi control functions

**Comment by gilgongo** [comment]
Hi @reneknuvers Thanks for this suggestion! Just so it gets some visibility and so that it can be discussed prior to possibly opening a PR (or a specification for a PR) I'll move this to the Discussions if that's OK.

**Comment by rdica** [comment]
I'm curious, does network xfer of midi data also suffer latency issues as well as audio?

**Reply by gilgongo** [reply]
I would assume so, but that's not necessarily a problem if audio latency isn't also a problem. 

In general though, I think (but I don't know) that this belongs to the set of ideas that are subject to constraints such as backward compatibility (if protocol changes are to be made), and the effect of loading non-audio data onto the UDP packet stream. So would not be easy to implement at all. As an aside, it's possible to do MIDI over RTP to achieve the same thing, although of course not as conveniently as if it were built into Jamlus.
